### PR TITLE
[MPS] Vectorize elementwise ops on inner-contiguous (sliced/strided) views

### LINKED ISSUE #188482: [MPS] TensorIterator should take advantage of contiguous inner dimension even if tensor is non-contiguous as a whole
### 🐛 Describe the bug

Follow-up to discussion at: https://github.com/pytorch/pytorch/pull/188200. 

Elementwise unary/binary ops and same-dtype copies on a tensor that is contiguous in its innermost dimension but non-contiguous overall fall back to the fully generic per-element strided kernel. For example

```python
 y = torch.exp(x[:, :W]) # non-contiguous argument -> hits the per-element kernel
```

Yet the innermost run is a clean unit stride span that could move as wide coalesced vectors. This leaves a lot of bandwidth on the table.

Opening a PR with a suggestion: a new kernel flavor that detects this case via TensorIterator's existing contiguity info and dispatches a vectorized path benefiting the elementwise metal kernels built on the iterator.

### Versions

nightly, M2 Max, macOS27 beta

cc @kulinseth @malfet @DenisVieriu97 @aditvenk @Isalia20

### PR DESCRIPTION
**Pull request by jhavukainen**
Fixes: https://github.com/pytorch/pytorch/issues/188482

This adds a new TensorIterator kernel flavor inner_contiguous. It is selected when the iterator is non-contiguous but its coalesced innermost dimension is unit-stride for every operand (TensorIterator::has_contiguous_first_dim()), ndim ≥ 2 and the inner extent clears a small threshold. 

The kernel computes each operand's outer offset once per contiguous run and then applies the existing ILP scheme (4 elements/thread) across the inner span to give coalesced accesses instead of per-element scalar traffic. When the inner dimension isn't contiguous it falls back to the strided kernel

For the special case of a same-dtype identity copy it dispatches one untyped kernel that moves the inner run as the widest aligned vector.

### Benchmarking
The op level S-curve for speedups for before/after attached below  as well as the script to reproduce the results using the PR branch. Tested using M2 Max on macOS27 beta. The ops tested there are:

**Unary:** neg, abs, sign, relu, exp, log, sqrt, rsqrt, sin, cos, tanh, sigmoid, erf, expm1, log1p, floor, trunc, reciprocal, i0, bitwise_not
  
**Binary:** mul, add, sub, maximum, minimum, eq, lt, remainder, fmod, div, fmax, pow, atan2, hypot, copysign, logaddexp, xlogy, gcd, bitwise_and, bitwise_or

**Copy:** copy_

speedup

[plot_speedup.py](https://github.com/user-attachments/files/29481639/plot_speedup.py)

### CODE REVIEW

**Comment by jhavukainen** [comment]
@claude review the changes

**Review by malfet — APPROVED** [review]
